calculator.rb

```diff
@@ -0,0 +1,86 @@
+#list of operators
+operators = %W[add + subtract - multiply * divide / modulo % power **]
+
+#options of mathmatical operation user can choose from
+puts "Welcome to the Calculator Program. Which operator would you like to use?
+ 1. Add(+)
+ 2. Subtract(-)
+ 3. Multiply(*)
+ 4. Divide(/)
+ 5. Power(**)
+ 6. Modulo(%)
+Please choose one operator(name or symbol)"
+command = gets.chomp.downcase
+if command == "/" || command == "divide"
+  puts "As a reminder you cannot divided by 0"
+end
+
+#prompt user unitl they enter the correct math operator
+until operators.include?(command)
+  puts "\n\nPlease re-enter correct operators add(+), subtract(-),
+  multiply(*), divide(/), power(**), modulo(%)"
+  command = gets.chomp
+end
+
+#prompt user unitl they enter a numaric number
+puts "Please input the numbers you want to calculate:"
+print "Number:"
+number = gets.chomp.to_f
+while number =~ /\D/
```

Comment:
This doesn't actually do anything because you've already converted the input to a float.

Also looking at reading in numbers from user input, this does not seem DRY.  Maybe you should make a method!